reaper/reaper.go

```diff
@@ -93,8 +95,12 @@ func (m *Monitor) Wait(c *exec.Cmd) (int, error) {
 
 func (m *Monitor) Register(pid int, c *Cmd) {
 	m.Lock()
+	defer m.Unlock()
+	if status, ok := m.unknown[pid]; ok {
+		c.ExitCh <- status
```

Comment:
Writing to a channel while holding a lock looks dangerous.  The `WaitPid` method requires acquiring this lock (then quickly releases) before reading from this channel. Would it be OK to delete, unlock, then send on the channel?